clang/include/clang/Frontend/FrontendActions.h

```diff
@@ -34,12 +34,17 @@ class InitOnlyAction : public FrontendAction {
 
 /// Preprocessor-based frontend action that also loads PCH files.
 class ReadPCHAndPreprocessAction : public FrontendAction {
+  llvm::function_ref<void(CompilerInstance &)> OnCI;
```

Comment:
What does `OnCI` mean? Is there a clearer name we could choose and/or could we document the role of this callback?

Also, how do we know function_ref is safe here?

Reply:
The function acts on `CompilerInstance`, so I figured `OnCI` fits. But I guess it's a bit too terse. How about `AdjustCI`? It's basically here to be able to tweak the `CompilerInstance` in a way that `CompilerInvocation` alone can't (and neither can this `FrontendAction` without some additional context).

Reply:
`function_ref` is safe here because `ReadPCHAndPreprocessAction ` is a stack variable in `DependencyScanningAction::runInvocation()` where it's clear the callback live long enough. I don't mind being on the safe side and turning this into `std::function` if you'd prefer.

Reply:
My concern is that nothing prevents using this frontend action from some other code where it wouldn't be safe.  Maybe we can make it a unique_function?

Reply:
Sounds good.